### ISSUE TITLE
Add growable vector

### ISSUE THREAD

**Issue by Boarders** [ollef/environment-bench#2]
It might be useful to add the relatively recent http://hackage.haskell.org/package/data-vector-growable which as far as I understand is roughly equivalent to std::vector (without a good story for deletion). It is similar to the package https://hackage.haskell.org/package/dynamic-mvector-0.1.0.5 that @AndrasKovacs wrote (perhaps for this sort of purpose).

**Comment by ollef** [comment]
Yes, that'd be interesting to see! It'd require some changes to the test setup to handle mutation. And I guess it then becomes relevant to include resetting the environment to what it was before adding some elements in the benchmark.

**Comment by AndrasKovacs** [comment]
It'd be kind of awkward for interpreter environments, because mutable env update is unsound in the presence of closures, so we'd still have to copy arrays frequently.

**Comment by ollef** [comment]
This is a good point. We probably need a more representative benchmark (e.g. an actual interpreter) for this to be fair.

I've also been thinking of creating a pure interface to an environment type that uses mutable, growable vectors under the hood, with something like copy-on-write when the same environment is extended multiple times.
